jep/200/README.adoc

```diff
@@ -276,7 +280,85 @@ For similar reasons, certain tests written in Groovy rather than Java prevent no
 
 == Reference Implementation
 
-link:https://github.com/jenkinsci/jenkins/pull/3120/files[Jenkins PR 3120] contains the bulk of the change and links to related PRs.
+* link:https://github.com/jenkinsci/jenkins/pull/3120[Jenkins PR #3120] contains the bulk of the change and links to related PRs.
+* link:https://github.com/jenkinsci/remoting/pull/208[Remoting PR #208] introduces the new API required to deliver the change.
+
+== Rollout plan
+
+=== Phase 1. Remoting release
+
+Target date: Dec 29, 2017
+
+JEP-200 requires extra API in Remoting, so link:https://github.com/jenkinsci/remoting/pull/208[pull request #208]
+needs to be delivered in advance.
+https://github.com/oleg-nenashev[@oleg-nenashev] will release the following components once the rollout plan is confirmed with the JEP sponsor:
+
+* Remoting
+* link:https://github.com/jenkinsci/docker-slave[docker-slave] (including the `latest` tag)
+* link:https://github.com/jenkinsci/docker-jnlp-slave[docker-jnlp-slave] images (including the `latest` tag)
+* link:https://github.com/jenkinsci/swarm-plugin/[Swarm Plugin Client]
+
+If everything goes fine, the Remoting release may be integrated into the weekly release
+(without actual use of the new API).
+
+=== Phase 2. Weekly Release
+
+Target date: Jan 13, 2017
+
+0. Once this JEP is approved, the https://jenkins.io/redirect/class-filter/ will be created on the Jenkins website
+** This document should provide a custom guide for creating JIRA issues with the `jep-200` label
+0. The JEP sponsor will write an announcement blogpost, which will describe the change and provide links to mitigation guidelines
+** This blog-post will be used as part of the upgrade guideline for LTS
+0. Once the blog post draft is approved,
+link:https://github.com/jenkinsci/jenkins/pull/3120[Jenkins PR #3120] will be integrated towards the next weekly release
+
+// oleg-nenashev: This process is a bit harsh for a single JEP sponsor, but in this case the sponsor represents a company AFAIK.
+// So I feel reasonable to set a maintenance quality bar taking a serious risk of regressions into account
+After the weekly release the JEP sponsor
+(or a group of people nominated by him, _JEP-200 maintainer(s)_)
+will be responsible to provide an extra support for the issues:
+
+* Issues with the `jep-200` label in Jenkins JIRA get first response within 2 business days (Monday-Friday)
+* Eligible components: Any non-blacklisted plugin hosted within the `jenkinsci` organization
+with more than `100` installations.
+* _JEP-200 maintainer(s)_ may request additional information from the reporter,
+but they are expected to provide a response within 2 business days after each iteration.
+Finally, they are expected to communicate the triage outcome.
+* Possible triage outcomes:
+** Accepted - patch in the plugin
+** Accepted - update whitelist in the core (similar to link:https://github.com/jenkinsci/jenkins/pull/3120/files#diff-ff24cb984ddd641f49a22cc13a90cfd3[these cases])
```

Comment:
Typed supplied _by_ the plugin should be whitelisted there. Types supplied by core, the Java Platform, etc. should be whitelisted in core, though the plugin may also add a whitelist entry as a stopgap.

Reply:
As this is getting closer to approval, I think it would make sense to redundantly file whitelist entries in plugins which currently ship something in the core whiteliste, namely:

```
hudson.maven.MavenInformation
javaposse.jobdsl.dsl.GeneratedJob
org.eclipse.jgit.lib.ObjectId
org.eclipse.jgit.lib.ObjectIdOwnerMap$Entry
org.eclipse.jgit.lib.PersonIdent
org.eclipse.jgit.revwalk.RevCommit
org.eclipse.jgit.revwalk.RevObject
org.eclipse.jgit.revwalk.RevTree
org.eclipse.jgit.transport.URIish
org.jboss.marshalling.TraceInformation$FieldInfo
org.jboss.marshalling.TraceInformation$ObjectInfo
```

Reply:
Maybe makes sense to have an EPIC in JIRA, so we could track that

Reply:
@jglick says there is a way to whitelist libraries by Manifest entries: https://github.com/jenkinsci/lib-jenkins-maven-embedder/pull/15/files. For particular classes it can be whitelisted by a resource file. It will need to be documented in the jenkins.io documentation page or in JEP so that plugin maintainers will be able to quickly whitelist things.

Reply:
Added an extensibility section to the JEP. SOLVED?